package dmMeshDDF;

import "ddf/ddf_extensions.proto";

option java_package = "com.dynamo.mesh.proto";
option java_outer_classname = "Mesh";
option (ddf_java_package) = "com.dynamo.mesh.ddf";

message MeshComponent
{
    required string    Name = 1;
    repeated float     Positions = 2;
    repeated float     Normals = 3;
    repeated float     Texcoord0 = 4;
    repeated uint32    Indices = 5;
    required uint32    PrimitiveCount = 6;
}

message MeshDesc
{
    enum Primitive
    {
        TRIANGLES = 1;
        QUADS = 2;
    }

    required Primitive PrimitiveType = 1;
    repeated MeshComponent Components = 2;
}
